--relocatable-device-code=true and --device-c (-dc) nvcc flags are not handled by cucc wrapper

When compiling CUDA programs with flags `--relocatable-device-code=true` or `--device-c` (`--device-c` is the same as `-dc` or `--relocatable-device-code=true --compile` according to [this CUDA documentation](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/#device-c-dc)), chipStar doesn't recognize the flags and instead directly use them for compiling without converting them to the corresponding clang flags (`-fgpu-rdc` flag for `--relocatable-device-code=true` and `-fgpu-rdc -c` for `--device-c` (`-dc`)).

[Reproducer]
Have the following three files, main.cu, function.hpp, and function.cu:

**main.cu**
```
#include "function.hpp"

int main() {
  test();
}
```
**function.hpp**
```
void test();
```
**function.cu**
```
#include "function.hpp"

void test() {
  printf("hello\n");
}
```
Then, compile main.cu first with `nvcc -dc main.cu`. The following errors will appear (with the -dc flag in the compilation command line):
```
/usr/bin/ld: /tmp/main-9ac6a2.o: in function `main':
main.cu:(.text+0x5): undefined reference to `test()'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)

failed to execute:/home/tsaini.chen/install/llvm/17.0/bin/clang++ -include /home/tsaini.chen/install/chipStar/chipStar_06272024/include/hip/spirv_fixups.h -I//home/tsaini.chen/install/chipStar/chipStar_06272024/include -isystem /home/tsaini.chen/install/chipStar/chipStar_06272024/include/cuspv -include /home/tsaini.chen/install/chipStar/chipStar_06272024/include/cuspv/cuda_runtime.h -D__NVCC__ -D__CHIP_CUDA_COMPATIBILITY__ -dc -x hip main.cu -D__HIP_PLATFORM_SPIRV__= --offload=spirv64 -nohipwrapperinc --hip-path=/home/tsaini.chen/install/chipStar/chipStar_06272024 --target=x86_64-unknown-linux-gnu   -L/home/tsaini.chen/install/chipStar/chipStar_06272024/lib -lCHIP -no-hip-rt -Wl,-rpath,/home/tsaini.chen/install/chipStar/chipStar_06272024/lib
```

--- a/bin/cucc.py
+++ b/bin/cucc.py
@@ -133,6 +133,12 @@ def prepare_argparser() -> argparse.ArgumentParser:
                         nargs=0, action=IgnoredOption)
     parser.add_argument("-gencode", "--gencode",
                         nargs="?", action=IgnoredOption)
+
+    # Add new arguments for relocatable device code and device compilation
+    parser.add_argument("--relocatable-device-code", choices=['true', 'false'])
+    parser.add_argument("--device-c", action='store_true')
+    parser.add_argument("-dc", action='store_true')
+
     return parser
 
 def get_hip_path() -> str:
@@ -275,6 +281,15 @@ def main():
             # confused by the extra output from the hipcc.
         else:
             hipcc_args.append("-v")
+
+    # Handle relocatable device code and device compilation flags
+    if (known_args.relocatable_device_code == 'true' or
+        known_args.device_c or
+        known_args.dc):
+            hipcc_args.append("-fgpu-rdc")
+    
+    if known_args.device_c or known_args.dc or "-dc" in other_args or "--device-c" in other_args:
+        hipcc_args.append("-c")
 
     if other_args is not None:
         hipcc_args.extend(filter_args_for_hipcc(other_args))

--- a/tests/cuda/CMakeLists.txt
+++ b/tests/cuda/CMakeLists.txt
@@ -90,3 +90,31 @@ add_compile_test(NAME cucc-include-cuda-runtime-twice
 #           cuda-inside.c' line correctly.
 #add_compile_test(NAME cucc-x-behavior
 #  COMPILE_OPTIONS "cuda-inside.cpp cuda-inside.c -x cu")
+
+# Test that cucc translates --relocatable-device-code=true to -fgpu-rdc
+add_compile_test(NAME cucc-relocatable-device-code-true
+  COMPILE_OPTIONS "--relocatable-device-code=true hello.cu -c -o /dev/null")
+set_tests_properties(cucc-relocatable-device-code-true PROPERTIES
+  ENVIRONMENT CHIP_CUCC_VERBOSE=1
+  PASS_REGULAR_EXPRESSION "Executing: .*hipcc .*-fgpu-rdc.*")
+
+# Test that cucc translates --device-c to -fgpu-rdc -c
+add_compile_test(NAME cucc-device-c
+  COMPILE_OPTIONS "--device-c hello.cu -o /dev/null")
+set_tests_properties(cucc-device-c PROPERTIES
+  ENVIRONMENT CHIP_CUCC_VERBOSE=1
+  PASS_REGULAR_EXPRESSION "Executing: .*hipcc .*-fgpu-rdc.*-c.*")
+
+# Test that cucc translates -dc to -fgpu-rdc -c
+add_compile_test(NAME cucc-dc
+  COMPILE_OPTIONS "-dc hello.cu -o /dev/null")
+set_tests_properties(cucc-dc PROPERTIES
+  ENVIRONMENT CHIP_CUCC_VERBOSE=1
+  PASS_REGULAR_EXPRESSION "Executing: .*hipcc .*-fgpu-rdc.*-c.*")
+
+# Test that cucc translates --relocatable-device-code=true --compile to -fgpu-rdc -c
+add_compile_test(NAME cucc-relocatable-device-code-true-compile
+  COMPILE_OPTIONS "--relocatable-device-code=true --compile hello.cu -o /dev/null")
+set_tests_properties(cucc-relocatable-device-code-true-compile PROPERTIES
+  ENVIRONMENT CHIP_CUCC_VERBOSE=1
+  PASS_REGULAR_EXPRESSION "Executing: .*hipcc .*-fgpu-rdc.*-c.*")

